engine/table/src/main/java/io/deephaven/engine/table/impl/sources/ReinterpretUtils.java

```diff
@@ -98,6 +99,68 @@ public static ColumnSource<Long> zonedDateTimeToLongSource(final @NotNull Column
         }
     }
 
+    /**
+     * Given a DateTime column source turn it into a long column source via reinterpretation if possible.
+     *
+     * @param source the source to turn into a long source
+     *
+     * @return the long source or null if it could not be reinterpretted
+     */
+    public static WritableColumnSource<?> writableDateTimeToLongSource(WritableColumnSource<?> source) {
+        if (source.allowsReinterpret(long.class)) {
+            return (WritableColumnSource<Long>) source.reinterpret(long.class);
+        }
+        return null;
+    }
+
+    /**
+     * Given a Boolean column source turn it into a byte column source via reinterpretation if possible.
+     *
+     * @param source the source to turn into a byte source
+     *
+     * @return the byte source or null if it could not be reinterpretted
+     */
+    public static WritableColumnSource<?> writableBooleanToByteSource(WritableColumnSource<?> source) {
+        if (source.allowsReinterpret(byte.class)) {
+            return (WritableColumnSource<Byte>) source.reinterpret(byte.class);
+        }
+        return null;
+    }
+
+    /**
+     * Given an {@link Instant} column source turn it into a long column source, either via reinterpretation or
+     * wrapping.
```

Comment:
Remove `or wrapping` from this one and `writableInstantToLongSource`.
Maybe consider mentioning that the source is writable on all 5 new methods? Feel not strictly necessary, since the types are specified.